### PR TITLE
API: Follow request - pass the "source" in the POST url not in the POST body

### LINKED ISSUE #8483: API: Follow requests pass source in POST Body JSON instead of Query string
### Expected behavior
Follows from the App are correctly attributed to ios instead on unknown

### Actual behavior

The code here: https://github.com/wordpress-mobile/WordPress-iOS/blob/15fa82b48ccf9e7eb5002e9654edf1d5138dca2a/WordPressKit/WordPressKit/ReaderSiteServiceRemote.m#L44
Passes the source as a Parameter which ends up in the POST body as JSON.

The API is defined as accepting source in the query string.

### Steps to reproduce the behavior
Follow a site in the App

##### Tested on [device], iOS [version], WPiOS [version]

WordPress/9.1 (iPhone; iOS 11.2.2; and by @nheagy in the emulator.

### PR DESCRIPTION
**Pull request by Rhetorical**
Fixes #8483

### CODE REVIEW

File: WordPressKit/WordPressKit/ReaderSiteServiceRemote.m [hunk 1]
```diff
@@ -43,13 +43,11 @@ - (void)fetchFollowedSitesWithSuccess:(void(^)(NSArray *sites))success failure:(
...
 {
-    NSString *path = [NSString stringWithFormat:@"sites/%lu/follows/new", (unsigned long)siteID];
+    NSString *path = [NSString stringWithFormat:@"sites/%lu/follows/new?%@=%@", (unsigned long)siteID, ReaderSiteServiceRemoteSourceKey, ReaderSiteServiceRemoteSourceValue];
     NSString *requestUrl = [self pathForEndpoint:path
                                      withVersion:ServiceRemoteWordPressComRESTApiVersion_1_1];
 
-    NSDictionary *params = @{ReaderSiteServiceRemoteSourceKey: ReaderSiteServiceRemoteSourceValue};
-
-    [self.wordPressComRestApi POST:requestUrl parameters:params success:^(id responseObject, NSHTTPURLResponse *httpResponse) {
+    [self.wordPressComRestApi POST:requestUrl parameters:@{} success:^(id responseObject, NSHTTPURLResponse *httpResponse) {
```
Comment (frosty) [inline WordPressKit/WordPressKit/ReaderSiteServiceRemote.m]: We could just pass `nil` for `parameters` here instead of constructing an empty array.

**Comment by Rhetorical** [comment]
@frosty thanks for the comment, I've updated the code.

**Comment by frosty** [comment]
Thanks for contributing!

**Comment by bummytime** [comment]
Nice work @Rhetorical. Thank you!

**Comment by Rhetorical** [comment]
glad I could help!

**Pull request merged**